odl/deform/linearized.py

```diff
@@ -76,8 +76,10 @@ def _linear_deform(template, displacement, out=None):
     """
     image_pts = template.space.points()
     for i, vi in enumerate(displacement):
-        image_pts[:, i] += vi.ntuple.asarray()
-    return template.interpolation(image_pts.T, out=out, bounds_check=False)
+        image_pts[:, i] += vi.asarray().ravel(order=vi.order)
+    values = template.interpolation(image_pts.T, out=out, bounds_check=False)
+    return values.reshape(displacement.space[0].shape,
+                          order=displacement.space[0].order)
```

Comment:
Reason?
Also, `DiscreteLp` has `default_dtype` not implemented since it depends on `impl`.

Perhaps `values.reshape(template.space.shape, order=template.space.order)` makes the intentions more clear? (They should all be the same anyway according to the doc.)

Reply:
> Also, DiscreteLp has default_dtype not implemented since it depends on impl.

That is, it cannot be implemented as `@staticmethod` as for the other classes.